examples/s2-vite-project/vite.config.ts

```diff
@@ -13,16 +13,28 @@
 import { defineConfig } from 'vite'
 import react from '@vitejs/plugin-react'
 import macros from 'unplugin-parcel-macros';
-import svgr from "vite-plugin-svgr";
 
 // https://vitejs.dev/config/
 export default defineConfig({
   plugins: [
     macros.vite(),
-    svgr({
-      svgrOptions: { exportType: 'named', ref: true, svgo: false, titleProp: true },
-      include: '**/*.svg',
-    }),
     react()
   ],
+  build: {
+    target: ['es2022'],
+    // Lightning CSS produces much a smaller CSS bundle than the default minifier.
+    cssMinify: 'lightningcss',
+    rollupOptions: {
+      output: {
+        // Bundle all S2 and style-macro generated CSS into a single bundle instead of code splitting.
+        // Because atomic CSS has so much overlap between components, loading all CSS up front results in
+        // smaller bundles instead of producing duplication between pages.
+        manualChunks(id) {
+          if (/macro-(.*)\.css$/.test(id) || /@react-spectrum\/s2\/.*\.css$/.test(id)) {
+            return 's2-styles';
+          }
+        }
+      }
+    }
```

Comment:
Just to make sure, this configuration meant to only apply to the built version of the app and not to local dev? I noticed the styles were still in separate style sheets when running local dev but not in the built version unlike the webpack one.

Reply:
Yeah that's right. Vite does not bundle during development, so unfortunately there's no way to deduplicate there. This applies to the built version only.